storage/innobase/include/ut0new.h

```diff
@@ -586,13 +609,23 @@ class ut_allocator {
 		pointer	ptr = reinterpret_cast<pointer>(
 			os_mem_alloc_large(&n_bytes));
 
+		if (set_to_zero)
+		{
+			::memset(ptr, 1, n_bytes);
+		}
+#if defined(DBUG_OFF) && defined(HAVE_MADVISE) && defined(MADV_DONTDUMP)
+		if (ptr && dontdump && madvise(ptr, n_bytes, MADV_DONTDUMP))
+                {
+			ib::warn() << "Failed to set memory to DONTDUMP: "
+                                                                << strerror(errno);
+                }
```

Comment:
Could you please define a common utility function for the ‘dontdump’ and ‘dodump’ calls? These calls are duplicated at least in 2 places.